codes/csharp/chapter_computational_complexity/time_complexity.cs

```diff
@@ -0,0 +1,277 @@
+				{
+					if (nums[j] > nums[j + 1])
+					{
+						// 交换 nums[j] 与 nums[j + 1]
+						int tmp = nums[j];
+						nums[j] = nums[j + 1];
+						nums[j + 1] = tmp;
+						count += 3;  // 元素交换包含 3 个单元操作
+					}
+				}
+			}
+			return count;
+		}
+
+		/// <summary>
+		/// 指数阶（循环实现）
+		/// </summary>
+		/// <param name="n"></param>
+		/// <returns></returns>
+		int exponential(int n)
+		{
+			int count = 0, baseNum = 1;
+			// cell 每轮一分为二，形成数列 1, 2, 4, 8, ..., 2^(n-1)
+			for (int i = 0; i < n; i++)
+			{
+				for (int j = 0; j < baseNum; j++)
+				{
+					count++;
+				}
+				baseNum *= 2;
+			}
+			// count = 1 + 2 + 4 + 8 + .. + 2^(n-1) = 2^n - 1
+			return count;
+		}
+
+		/// <summary>
+		/// 指数阶（递归实现）
+		/// </summary>
+		/// <param name="n"></param>
+		/// <returns></returns>
+		int expRecur(int n)
+		{
+			if (n == 1) return 1;
+			return expRecur(n - 1) + expRecur(n - 1) + 1;
+		}
+
+		/// <summary>
+		/// 对数阶（循环实现）
+		/// </summary>
+		/// <param name="n"></param>
+		/// <returns></returns>
+		int logarithmic(float n)
+		{
+			int count = 0;
+			while (n > 1)
+			{
+				n = n / 2;
+				count++;
+			}
+			return count;
+		}
+
+		/// <summary>
+		/// 对数阶（递归实现）
+		/// </summary>
+		/// <param name="n"></param>
+		/// <returns></returns>
+		int logRecur(float n)
+		{
+			if (n <= 1) return 0;
+			return logRecur(n / 2) + 1;
+		}
+
+		/// <summary>
+		/// 线性对数阶
+		/// </summary>
+		/// <param name="n"></param>
+		/// <returns></returns>
+		int linearLogRecur(float n)
+		{
+			if (n <= 1) return 1;
+			int count = linearLogRecur(n / 2) +
+						linearLogRecur(n / 2);
+			for (int i = 0; i < n; i++)
+			{
+				count++;
+			}
+			return count;
+		}
+
+		/// <summary>
+		/// 阶乘阶（递归实现）
+		/// </summary>
+		/// <param name="n">递归数</param>
+		/// <returns></returns>
+		int factorialRecur(int n)
+		{
+			if (n == 0) return 1;
+			int count = 0;
+			// 从 1 个分裂出 n 个
+			for (int i = 0; i < n; i++)
+			{
+				count += factorialRecur(n - 1);
+			}
+			return count;
+		}
+
+		/// <summary>
+		/// 生成一个数组，元素为 { 1, 2, ..., n }，顺序被打乱
+		/// </summary>
+		/// <param name="n">数组大小</param>
+		/// <returns></returns>
+		int[] randomNumbers(int n)
+		{
+			int[] nums = new int[n];
+			// 生成数组 nums = { 1, 2, 3, ..., n }
+			for (int i = 0; i < n; i++)
+			{
+				nums[i] = i + 1;
+			}
+			// 随机打乱数组元素
+			nums = nums.OrderBy(num => System.Random.Shared.Next()).ToArray();
+			return nums;
+		}
+
+		/// <summary>
+		/// 查找数组 nums 中数字 1 所在索引
+		/// </summary>
+		/// <param name="nums">索引数组</param>
+		/// <returns></returns>
+		int findOne(in Span<int> nums) => nums.IndexOf(1);
+
+		void worstBestTimeComplexity()
+		{
+			for (int i = 0; i < 10; i++)
+			{
+				int n = 100;
+				int[] nums = randomNumbers(n);
+				int index = findOne(nums);
+				System.Console.WriteLine($"打乱后的数组为 [{string.Join(",", nums)}]");
+				System.Console.WriteLine($"数字 1 的索引为 [{index}]");
+			}
+		}
+
+		/// <summary>
+		/// Driver Code
+		/// </summary>
+		public void main()
```

Comment:
We use `using NUnit.Framework` to do the unit test. Please refer to `csharp/array_and_linkedlist/Array.cs`